src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -0,0 +1,335 @@
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.ParallelDeadlineGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.StartEndCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import frc.robot.maps.RobotMap.IndexMap;
+
+/**
+ * The indexer lines up and transports the balls to the Shooter The indexer has
+ * two modes, On(motors on) and Off(motors off) The indexer relates to the
+ * Shooter in that it should only put the balls in when the Shooter is ready The
+ * indexer is triggered with motors and uses belts to move the balls(controller)
+ * The indexer will use IR sensors so we can see the presence of a ball The
+ * distance that the balls are pushed between the belts needs to be coded so
+ * that they can move fluidly When a ball is at the top of the Pierre, the ball
+ * at the entrance of Pierre needs to be moved forward at the same pace of the
+ * balls being intaked
+ * 
+ */
+
+public class Indexer extends SubsystemBase {
+    final SendableSpeedController singulator;
+    private final SendableSpeedController pierreMotor;
+    final BooleanSupplier frontIntakeIR;
+    final BooleanSupplier bottomPierreIR;
+    final BooleanSupplier topPierreIR;
+    final BooleanSupplier backIntakeIR;
+    public double ballCounting;
+
+    private static final double singulatorMotorSpeed = 0.95;
+    private static final double pierreIndexSpeed = 0.85;
+
+
+
+    public Indexer(final IndexMap map) {
+        super();
+        singulator = map.singulator();
+        frontIntakeIR = map.frontIntakeIR();
+        bottomPierreIR = map.bottomPierreIR();
+        topPierreIR = map.topPierreIR();
+        backIntakeIR = map.backIntakeIR();
+
+        // bottomPierreIR = map.bottomPierreIR();
+        // topPierreIR = map.topPierreIR();
+        // backIntakeIR = map.backIntakeIR();
+        pierreMotor = map.pierreMotor();
+        
+    }
+
+    public SequentialCommandGroup intakeToPierre() {
+        return new SequentialCommandGroup(pierrePossesion(), runToClear());
+    }
+    
+    public SequentialCommandGroup shootOneBall() {
+        return new SequentialCommandGroup(loadBallToTop(), unLoadBall());
+    }
+
+    public ParallelDeadlineGroup topSensorTriggered(){
+        final ParallelDeadlineGroup topIRStop = new ParallelDeadlineGroup(intakeToPierre(), pierrePossesion(),
+                singulatorPossesion(),
+                loadBallToTop(), runToClear());
+
+
+         topIRStop.setDeadline(stopWhenBallsAtTop());
+
+         return topIRStop;
+    }
+    //This will stop all commands when the top IR sensor on Pierre is triggered
+
+    public CommandBase indexMotor(final double motorSpeed) {
+        return new StartEndCommand(() -> {
+            singulator.set(motorSpeed);
+        }, () -> {
+            singulator.set(0);
+        });
+    }
+
+    public CommandBase runPierre() {
```

Comment:
This isn't used in this file, and should probably be removed. We shouldn't need a command like this.

Reply:
This is literally the same thing as `indexMotor(.75)`... shouldn't be necessary.